src/main/java/org/jvnet/hudson/plugins/repositoryconnector/VersionParameterDefinition.java

```diff
@@ -53,35 +58,74 @@ public VersionParameterDefinition(String repoid, String groupid,
     public VersionParameterDefinition copyWithDefaultValue(ParameterValue defaultValue) {
         if (defaultValue instanceof StringParameterValue) {
             // TODO: StringParameterValue value = (StringParameterValue) defaultValue;
-            return new VersionParameterDefinition(getRepoid(), "",
-                    "", "", getDescription());
+            return new VersionParameterDefinition(getRepoid(), getGroupid(), getArtifactid(), "", getDescription(),true);
         } else {
             return this;
         }
     }
 
+    public boolean reverseOrder() {
+    	return reverseOrder;
+    }
+    
+    public Collection<Repository> getRepos(String repoid) {
+    	List<Repository> out = new ArrayList<Repository>();
+    	
+    	for(Map.Entry<String,Repository> e : RepositoryConfiguration.get().getRepositoryMap().entrySet()) {
+    		if(repoid.compareTo("ALL")==0 || repoid.compareTo(e.getKey())==0) {
+    			out.add(e.getValue());
+    		}
+    	}
+    	return out;
+    }
+    
+    
     @Exported
     public List<String> getChoices() {
-        Repository r = DESCRIPTOR.getRepo(repoid);
-        List<String> versionStrings = new ArrayList<String>();
-        if (r != null) {
-            File localRepo = RepositoryConfiguration.get().getLocalRepoPath();
-            Aether aether = new Aether(DESCRIPTOR.getRepos(), localRepo);
-            try {
-                List<Version> versions = aether.resolveVersions(groupid, artifactid);
-                for (Version version : versions) {
-                    versionStrings.add(version.toString());
-                }
-            } catch (VersionRangeResolutionException ex) {
-                log.log(Level.SEVERE, "Could not determine versions", ex);
-            }
-            if (!versionStrings.isEmpty()) {
-                // reverseorder to have the latest versions on top of the list
-                Collections.reverse(versionStrings);
-                // add the default parameters
-                versionStrings.add(0, "LATEST");
-                versionStrings.add(0, "RELEASE");
+    	Collection<Repository> repos = getRepos(repoid);
+
+    	List<String> versionStrings = new ArrayList<String>();
+
+    	File localRepo = RepositoryConfiguration.get().getLocalRepoPath();
+        log.info("VersionParameterDefinition: local repo "+localRepo.getAbsolutePath());
+        
+        Aether aether = new Aether(
+        		repos, localRepo, null, false, 
+        		RepositoryPolicy.UPDATE_POLICY_ALWAYS, 
+                RepositoryPolicy.CHECKSUM_POLICY_FAIL, 
```

Comment:
well - why don't we return a useful error message then? in general, if your checksums fail, you're in deep shit anyway. in a proper setup, this shouldn't happen. with a warning, we would have the same issue as with and error: in order to make sure people actually notice the warning ... it would have to be reported nicely. my feeling is that not acting on checksum errors is the repo equivalent of accepting any self-signed certificate in an ssl connection ... not a good idea